### PR TITLE
feat(new sink): Initial basic `datadog_logs` sink

### LINKED ISSUE #1469: New `datadog_logs` sink
A sink that sends log events to [Datadog's log service](https://docs.datadoghq.com/logs/).

- I'm wondering if this could be easily solved with #1434.
- This is blocking co-marketing with Datadog.

### PR DESCRIPTION
**Pull request by LucioFranco**
This implements a very basic datadog logs sink that will stream data via the tcp sink. At the moment it doesn't have many features compared to other agents but can properly stream text/json logs securely to datadog.

Closes #1469

### CODE REVIEW

File: src/sinks/datadog/logs.rs
Comment (bruceg) [inline src/sinks/datadog/logs.rs]: Should these be proper `Snafu` type errors?

File: src/sinks/datadog/logs.rs
Comment (bruceg) [inline src/sinks/datadog/logs.rs]: Would a `map` be simpler here?

`util::encode_event(event, encoding).map(move |bytes| { api_key.extend(bytes); api_key })`

**Review by bruceg — APPROVED** [review]
Looks ok to me, but I left a couple of questions.

File: src/sinks/datadog/logs.rs
Comment (LucioFranco) [reply src/sinks/datadog/logs.rs]: I figured for such a simple sink its probably fine to just use the string here.

File: src/sinks/datadog/logs.rs
Comment (LucioFranco) [reply src/sinks/datadog/logs.rs]: I would if there was more in this fn but I find something simple to be a good use case for this type of control flow.

File: website/docs/reference/sinks/datadog_logs.md.erb
Comment (binarylogic) [inline website/docs/reference/sinks/datadog_logs.md.erb]: Please add a section "Obtaining a Datadog API Key"

File: .meta/sinks/datadog_logs.toml
Comment (binarylogic) [inline .meta/sinks/datadog_logs.toml]: Please remove the `category` field.

File: .meta/sinks/datadog_logs.toml
Comment (binarylogic) [inline .meta/sinks/datadog_logs.toml]: Is this common? Are users routinely setting this?

File: .meta/sinks/datadog_logs.toml
Comment (binarylogic) [inline .meta/sinks/datadog_logs.toml]: Shouldn't this default to `json`?

File: .meta/sinks/datadog_logs.toml
Comment (binarylogic) [inline .meta/sinks/datadog_logs.toml]: Please link to Datadog's log API.

**Review by binarylogic — CHANGES_REQUESTED** [review]
A few changes above. Also, are we mapping our own internal schema to theirs? All of the fields listed  [here](https://docs.datadoghq.com/logs/processing/#reserved-attributes).

**Review by lukesteensen — APPROVED** [review]

File: .meta/sinks/datadog_logs.toml
Comment (Hoverbear) [reply .meta/sinks/datadog_logs.toml]: Since #1915 we don't default when there is more than one option, based on @binarylogic 's advice.

File: website/docs/reference/sinks/datadog_logs.md.erb
Comment (binarylogic) [inline website/docs/reference/sinks/datadog_logs.md.erb]: Why `fluentd`? Is there not a generic HTTP or curl option?

Comment (LucioFranco) [reply website/docs/reference/sinks/datadog_logs.md.erb]: Not that I could find at all and it seemed fluentd/fluentbit had the easier config to parse/read and extract the api key. I think if we want to avoid this we must get vector in there.

File: website/docs/reference/sinks/datadog_logs.md.erb
Comment (Hoverbear) [inline website/docs/reference/sinks/datadog_logs.md.erb]:
```suggestion
### Obtaining an API token
```

**Review by Hoverbear — APPROVED** [review]

**Comment by Hoverbear** [comment]
@LucioFranco please fixup this PR

**Comment by LucioFranco** [comment]
@Hoverbear will do, I think I am still waiting on a doc review from @binarylogic

**Review by binarylogic — APPROVED** [review]

**Comment by binarylogic** [comment]
From planning: just needs to rebase.

**Pull request merged**
